File: app/src/main/java/com/example/quizmaster/LoginActivity.java
package com.example.quizmaster;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.content.Intent;
import android.text.TextUtils;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.Toast;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Button;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.AuthResult;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;

public class LoginActivity extends AppCompatActivity {

    private EditText emailTextView, passwordTextView;
    private Button Btn;
    private ProgressBar progressbar;
    private TextView tapHear;

    private FirebaseAuth fAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        // taking instance of FirebaseAuth
        fAuth = FirebaseAuth.getInstance();

        // initialising all views through id defined above
        emailTextView = findViewById(R.id.email_l);
        passwordTextView = findViewById(R.id.password_l);
        Btn = findViewById(R.id.login);
        progressbar = findViewById(R.id.progressBarOfLogin);
        tapHear = findViewById(R.id.tap_hear_to_register);

        tapHear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(),RegisterActivity.class));
            }
        });

        // Set on Click Listener on Sign-in button
        Btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v)
            {
                String email = emailTextView.getText().toString().trim();
                String password = passwordTextView.getText().toString().trim();

                if (TextUtils.isEmpty(email)) {
                    emailTextView.setError("Email required");
                    return;
                }

                if (TextUtils.isEmpty(password)) {
                    passwordTextView.setError("Password required");
                    return;
                }

                progressbar.setVisibility(View.VISIBLE);

                //sign in

                fAuth.signInWithEmailAndPassword(email,password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {

                        if (task.isSuccessful()) {
                            Toast.makeText(getApplicationContext(),
                                    "Login successful!",
                                    Toast.LENGTH_LONG)
                                    .show();
                            // if the user created intent to login activity
                            startActivity(new Intent(getApplicationContext(),CategoriesActivity.class));
                        }
                        else{

                            Toast.makeText(getApplicationContext(),
                                    "Login failed",
                                    Toast.LENGTH_LONG)
                                    .show();

                        }


                    }
                });


            }
        });
    }

}

File: app/src/main/java/com/example/quizmaster/LoginActivity.kt
package com.example.quizmaster

import android.content.Intent
import android.os.Bundle
import android.util.Patterns
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.auth.ktx.auth
import com.google.firebase.ktx.Firebase
import kotlinx.android.synthetic.main.activity_login.*
import kotlinx.android.synthetic.main.activity_register.*

class LoginActivity : AppCompatActivity() {

    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        auth = Firebase.auth

        register_first.setOnClickListener{
            val intent = Intent(this, RegistrationActivity::class.java)
            startActivity(intent)
        }

        login_done.setOnClickListener {
            this.doLogin()
        }

    }

    private fun doLogin(){
        if(Email.text.toString().isEmpty()){
            Email.error = "please enter a email"
            Email.requestFocus()
            return
        }

        if(!Patterns.EMAIL_ADDRESS.matcher(Email.text.toString()).matches()){
            Email.error = "enter valid email"
            Email.requestFocus()
            return
        }

        if(Password.text.toString().isEmpty()){
            Password.error = "please enter a password"
            Password.requestFocus()
            return
        }

        auth.signInWithEmailAndPassword(Email.text.toString(), Password.text.toString())
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    val user = auth.currentUser
                    updateUI(user)
                } else {
                    Toast.makeText(baseContext, "Authentication failed.",
                        Toast.LENGTH_SHORT).show()
                    updateUI(null)

                }

                // ...
            }
    }

    public override fun onStart() {
        super.onStart()
        // Check if user is signed in (non-null) and update UI accordingly.
        val currentUser = auth.currentUser
        updateUI(currentUser)
    }

    private fun updateUI(currentUser: FirebaseUser?) {
        if (currentUser != null){
            Toast.makeText(baseContext, "Login successful.",
                Toast.LENGTH_SHORT).show()
        }else{
            Toast.makeText(baseContext, "Login failed.",
                Toast.LENGTH_SHORT).show()
        }
    }
}